src/lug/Graphics/Vulkan/Gui.cpp

```diff
@@ -579,64 +579,65 @@ bool Gui::endFrame(const std::vector<VkSemaphore>& waitSemaphores, uint32_t curr
     return true;
 }
 
-void Gui::processEvent(const lug::Window::Event event) {
+bool Gui::processEvent(const lug::Window::Event event) {
     ImGuiIO& io = ImGui::GetIO();
-    switch (event.type) {
-        case lug::Window::Event::Type::KeyPressed:
-        case lug::Window::Event::Type::KeyReleased:
+
+    if (io.WantCaptureMouse) {
+        if (event.type == lug::Window::Event::Type::ButtonPressed ||
+            event.type == lug::Window::Event::Type::ButtonReleased) {
+            switch (event.mouse.code) {
+                case lug::Window::Mouse::Button::Left:
+                    io.MouseDown[0] = (event.type == lug::Window::Event::Type::ButtonPressed) ? true : false;
+                    break;
+                case lug::Window::Mouse::Button::Right:
+                    io.MouseDown[1] = (event.type == lug::Window::Event::Type::ButtonPressed) ? true : false;
+                    break;
+                case lug::Window::Mouse::Button::Middle:
+                    io.MouseDown[2] = (event.type == lug::Window::Event::Type::ButtonPressed) ? true : false;
+                    break;
+                case lug::Window::Mouse::Button::XButton1:
+                    io.MouseDown[3] = (event.type == lug::Window::Event::Type::ButtonPressed) ? true : false;
+                    break;
+                case lug::Window::Mouse::Button::XButton2:
+                    io.MouseDown[4] = (event.type == lug::Window::Event::Type::ButtonPressed) ? true : false;
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        return true;
+    } else if (io.WantCaptureKeyboard) {
```

Comment:
Isn't it possible for `io.WantCaptureMouse` and `io.WantCaptureKeyboard` to be `true` at the same time?